api/python/notebooks/experimental/pytorch_loader_scvi.ipynb

```diff
@@ -0,0 +1,560 @@
+    "\n",
+    "For this notebook, we'll only do a single epoch, but this should likely be increased. The SCVI models hosted in CELLxGENE have been trained for 100 epochs."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "GPU available: True (mps), used: False\n",
+      "TPU available: False, using: 0 TPU cores\n",
+      "IPU available: False, using: 0 IPUs\n",
+      "HPU available: False, using: 0 HPUs\n",
+      "/opt/cellxgene-census/venv/lib/python3.10/site-packages/lightning/pytorch/trainer/setup.py:187: GPU available but not used. You can set it by doing `Trainer(accelerator='gpu')`.\n",
+      "/opt/cellxgene-census/venv/lib/python3.10/site-packages/lightning/pytorch/trainer/connectors/data_connector.py:441: The 'train_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=11` in the `DataLoader` to improve performance.\n",
+      "/opt/cellxgene-census/venv/lib/python3.10/site-packages/lightning/pytorch/utilities/data.py:121: Your `IterableDataset` has `__len__` defined. In combination with multi-process data loading (when num_workers > 1), `__len__` could be inaccurate if each worker is not configured independently to avoid having duplicate data.\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Epoch 1/1:   0%|          | 0/1 [00:00<?, ?it/s]"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/opt/cellxgene-census/venv/lib/python3.10/site-packages/lightning/pytorch/utilities/data.py:77: Trying to infer the `batch_size` from an ambiguous collection. The batch size we found is 1024. To avoid any miscalculations, use `self.log(..., batch_size=batch_size)`.\n",
+      "/opt/cellxgene-census/venv/lib/python3.10/site-packages/lightning/pytorch/utilities/data.py:77: Trying to infer the `batch_size` from an ambiguous collection. The batch size we found is 903. To avoid any miscalculations, use `self.log(..., batch_size=batch_size)`.\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Epoch 1/1: 100%|██████████| 1/1 [01:51<00:00, 111.59s/it, v_num=1, train_loss_step=2.39e+3, train_loss_epoch=2.94e+3]"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "`Trainer.fit` stopped: `max_epochs=1` reached.\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Epoch 1/1: 100%|██████████| 1/1 [01:51<00:00, 111.60s/it, v_num=1, train_loss_step=2.39e+3, train_loss_epoch=2.94e+3]\n"
+     ]
+    }
+   ],
+   "source": [
+    "batch_size = 1024\n",
+    "train_size = 0.9\n",
+    "max_epochs = 1\n",
+    "\n",
+    "model.train(\n",
+    "    datamodule=datamodule,\n",
```

Comment:
Typo? should it be `data_module`?

Otherwise I get an error

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[6], line 5
      2 train_size = 0.9
      3 max_epochs = 1
----> 5 model.train(
      6     datamodule=datamodule,
      7     max_epochs=max_epochs,
      8     batch_size=batch_size,
      9     train_size=train_size,
     10     early_stopping=False,
     11 )

File ~/venv_uce/lib/python3.11/site-packages/scvi/model/base/_training_mixin.py:90, in UnsupervisedTrainingMixin.train(self, max_epochs, accelerator, devices, train_size, validation_size, shuffle_set_split, load_sparse_tensor, batch_size, early_stopping, datasplitter_kwargs, plan_kwargs, data_module, **trainer_kwargs)
     86     raise ValueError(
     87         "Cannot pass in `data_module` if the model was initialized with `adata`."
     88     )
     89 elif data_module is None and self._module_init_on_train:
---> 90     raise ValueError(
     91         "If the model was not initialized with `adata`, a `data_module` must be passed in."
     92     )
     94 if max_epochs is None:
     95     if data_module is None:

ValueError: If the model was not initialized with `adata`, a `data_module` must be passed in.
```

Reply:
Can't reproduce after fixing typo above I get the following error after the epoch is completed

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[7], line 5
      2 train_size = 0.9
      3 max_epochs = 1
----> 5 model.train(
      6     data_module=datamodule,
      7     max_epochs=max_epochs,
      8     batch_size=batch_size,
      9     train_size=train_size,
     10     early_stopping=False,
     11 )

File ~/venv_uce/lib/python3.11/site-packages/scvi/model/base/_training_mixin.py:143, in UnsupervisedTrainingMixin.train(self, max_epochs, accelerator, devices, train_size, validation_size, shuffle_set_split, load_sparse_tensor, batch_size, early_stopping, datasplitter_kwargs, plan_kwargs, data_module, **trainer_kwargs)
    131 trainer_kwargs[es] = (
    132     early_stopping if es not in trainer_kwargs.keys() else trainer_kwargs[es]
    133 )
    134 runner = self._train_runner_cls(
    135     self,
    136     training_plan=training_plan,
   (...)
    141     **trainer_kwargs,
    142 )
--> 143 return runner()

File ~/venv_uce/lib/python3.11/site-packages/scvi/train/_trainrunner.py:102, in TrainRunner.__call__(self)
     99 self._update_history()
    101 # data splitter only gets these attrs after fit
--> 102 self.model.train_indices = self.data_splitter.train_idx
    103 self.model.test_indices = self.data_splitter.test_idx
    104 self.model.validation_indices = self.data_splitter.val_idx

AttributeError: 'CensusSCVIDataModule' object has no attribute 'train_idx'
```